contracts/contracts/coordination/Subscription.sol

```diff
@@ -0,0 +1,261 @@
+     */
+    event SubscriptionPaid(
+        uint32 indexed subscriptionId,
+        address indexed subscriber,
+        uint256 amount
+    );
+
+    /**
+     * @notice Sets the coordinator and fee token contracts
+     * @dev The coordinator and fee token contracts cannot be zero addresses
+     * @param _coordinator The address of the coordinator contract
+     * @param _feeToken The address of the fee token contract
+     * @param _beneficiary The address of the beneficiary
+     */
+    constructor(Coordinator _coordinator, IERC20 _feeToken, address _beneficiary) {
+        require(address(_coordinator) != address(0), "Coordinator cannot be the zero address");
+        require(address(_feeToken) != address(0), "Fee token cannot be the zero address");
+        require(_beneficiary != address(0), "Beneficiary cannot be the zero address");
+        coordinator = _coordinator;
+        feeToken = _feeToken;
+        beneficiary = _beneficiary;
+    }
+
+    /**
+     * @notice Returns the subscription fee
+     * @return The subscription fee
+     */
+    function subscriptionFee() public pure returns (uint256) {
+        return 42 * 10 ** 20; // TODO
+    }
+
+    /**
+     * @notice Returns the base expiration duration
+     * @return The base expiration duration
+     */
+    function baseExpiration() public pure returns (uint256) {
+        return 52 weeks; // TODO
+    }
+
+    /**
+     * @notice Returns the key used to lookup authorizations
+     * @param ritualId The ID of the ritual
+     * @param encryptor The address of the encryptor
+     * @return The key used to lookup authorizations
+     */
+    function lookupKey(uint32 ritualId, address encryptor) internal pure returns (bytes32) {
+        return keccak256(abi.encodePacked(ritualId, encryptor));
+    }
+
+    /**
+     * @notice Creates a new subscription
+     * @param ritualId The ID of the ritual
+     */
+    function newSubscription(uint32 ritualId) external {
+        uint32 subscriptionId = numberOfSubscriptions;
+        SubscriptionInfo storage sub = subscriptions[subscriptionId];
+        sub.subscriber = msg.sender;
+        paySubscriptionFor(subscriptionId);
+
+        subscribers[lookupKey(ritualId, msg.sender)] = subscriptionId;
+        numberOfSubscriptions += 1;
+
+        emit SubscriptionCreated(subscriptionId, msg.sender, ritualId);
+    }
+
+    /**
+     * @notice Pays for a subscription
+     * @param subscriptionId The ID of the subscription
+     */
+    function paySubscriptionFor(uint32 subscriptionId) public virtual {
+        uint256 amount = subscriptionFee();
+
+        SubscriptionInfo storage sub = subscriptions[subscriptionId];
+        sub.paidFor += amount;
+        sub.expiration += baseExpiration();
+
+        feeToken.safeTransferFrom(msg.sender, address(this), amount);
+
+        // TODO: We already emit SubscriptionCreated, do we need this?
+        emit SubscriptionPaid(subscriptionId, msg.sender, amount);
```

Comment:
I'm for keeping this event, because method is public and I suppose it can be called outside without `SubscriptionCreated`